R/location.R

```diff
@@ -96,6 +96,21 @@ orderly_location_add <- function(name, type, args, verify = TRUE, root = NULL) {
 
   if (type == "path") {
     assert_scalar_character(args$path, name = "path")
+    if (!fs::is_absolute_path(args$path)) {
+      ## This only happens where the current working directory is not
+      ## the same as the root.
+      root_error <- !file.exists(file.path(root$path, args$path)) &&
+        file.exists(args$path)
+      if (root_error) {
+        root_fix <- as.character(fs::path_rel(args$path, root$path))
+        cli::cli_abort(
+          c("'path' must be given relative to the orderly root",
+            x = paste("You have provided the relative path '{args$path}',",
+                      "which exists, but does not exist relatively to",
+                      "'{root$path}', the root of your orderly archive"),
+            i = "Consider passing '{root_fix}' instead"))
+      }
```

Comment:
This is really nice diagnostics! Really solves the problem of having to choose between two equally acceptable behaviours.